hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerStateManagerImpl.java

```diff
@@ -438,14 +450,14 @@ public ContainerInfo getMatchingContainer(final long size, String owner,
     final ContainerID lastID =
         lastUsedMap.getOrDefault(key, containerIDs.first());
 
+    lockManager.readLock(lastID);
```

Comment:
`findContainerWithSpace` iterates containers, so I think we should not acquire lock for a single container ID here.  How about locking in `findContainerWithSpace` instead?

Reply:
Yes , makes sense. I will take care for taking lock at container level for each container inside findContainerWithSpace.

Reply:
And I agree that RW-lock and container level RW-lock work mutually exclusive, but if you see that `contains` method is picking container info from a concurrent hash map so no 2 threads will interfere to pick and `addContainer` is using same hashmap to add new container